### PR TITLE
"Element with `lang` attribute has valid language tag" (de46e4): Apply to body and ignore attributes with no effect

### LINKED ISSUE #1520: "Element with `lang` attribute has valid language tag" (de46e4): Should it check accessible names?
[Element with lang attribute has valid language tag](https://act-rules.github.io/rules/de46e4) only checks element with visible/accessible text nodes. This means that
```html
<div lang="invalid"><img src="…" alt="lorem ipsum" /></div>
```
is not tested because it has no descendant text nodes.

It's not a problem per se but could be an improvement (or a companion rule) to also check elements with descendants having an accessible name.

### LINKED ISSUE #1525: "Element with `lang` attribute has valid language tag" (de46e4) fails on nested attribute
[Element with `lang` attribute has valid language tag](https://act-rules.github.io/rules/de46e4) is currently failing on the `div` of
```html
<body>
  <div lang="invalid">
    <span lang="en">Hello World</span>
  </div>
</body>
```

Suggestion: wait for #1521 to be finished and use [text with the same language](https://github.com/act-rules/act-rules.github.io/blob/1894105b09818d55dfda9e19bcbbf28e5f29fc51/pages/glossary/text-same-language.md) to replace the 3rd point in Applicability by something like
> the element has visible/included text with the same language

### PR DESCRIPTION
**Pull request by Jym77**
Align the "lang valid" rule with the newer "lang matches default" one by
- also applying on `body` itself;
- only checking `lang` attributes who affect some text.

Closes issue(s):

- closes #1520 
- closes #1525 

Need for Final Call:
This will require a 1 week Final Call (small fixes to Applicability)

---

### **When creating PR:**

- [X] Make sure you're requesting to **pull a branch** (right side) to the `develop` branch (left side).
- [X] Make sure you do not remove the "How to Review and Approve" section in your pull request description

### **After creating PR:**

- [X] Add yourself (and co-authors) as "Assignees" for PR.
- [X] Add label to indicate if it's a `Rule`, `Definition` or `Chore`.
- [X] [Link the PR to any issue it solves](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue). This will be done automatically by [referencing the issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) at the top of this comment in the indicated place.
- [X] Optionally request feedback from anyone in particular by assigning them as "Reviewers".

### **When merging a PR:**

- [ ] Close any issue that the PR resolves. This will happen automatically upon merging if the PR was correctly linked to the issue, e.g. by referencing the issue at the top of this comment.

## How to Review And Approve

- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines.
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.
- Make sure to also review the proposed Final Call period. In case of disagreement, the longer period wins.

### CODE REVIEW

**Review by ajanec01 — APPROVED** [review]

**Review by WilcoFiers — APPROVED** [review]

File: _rules/element-lang-valid-de46e4.md [hunk 1]
```diff
@@ -29,9 +29,9 @@ acknowledgments:
 
 This rules applies to any HTML element with a `lang` [attribute value][] that is not empty (`""`) and for which all of the following is true:
 
-- the element is a [descendant][] in the [flat tree][] of a `body` element; and
-- the element has a [node document][] with a [content type][] of `text/html`; and
-- the element has a [text node][] as a [descendant][] in the [flat tree][] that is [visible][] or [included in the accessibility tree][].
+- **descendant** the element is an [inclusive descendant][] in the [flat tree][] of a `body` element; and
```
Comment (carlosapaduarte) [inline _rules/element-lang-valid-de46e4.md:32]:
```suggestion
- **descendant**: the element is an [inclusive descendant][] in the [flat tree][] of a `body` element; and
```

File: _rules/element-lang-valid-de46e4.md [hunk 2]
```diff
@@ -29,9 +29,9 @@ acknowledgments:
...
+- **content type** the element has an associated [node document][] with a [content type][] of `text/html`; and
```
Comment (carlosapaduarte) [inline _rules/element-lang-valid-de46e4.md:33]:
```suggestion
- **content type**: the element has an associated [node document][] with a [content type][] of `text/html`; and
```

File: _rules/element-lang-valid-de46e4.md [hunk 3]
```diff
@@ -29,9 +29,9 @@ acknowledgments:
...
+- **text** there is some non-empty [text inheriting its programmatic language][] from the element.
```
Comment (carlosapaduarte) [inline _rules/element-lang-valid-de46e4.md:34]:
```suggestion
- **text**: there is some non-empty [text inheriting its programmatic language][] from the element.
```

**Review by carlosapaduarte — CHANGES_REQUESTED** [review]
Some colons missing

**Review by carlosapaduarte — APPROVED** [review]

**Comment by Jym77** [comment]
Call for review ends on June 17th.

**Comment by Jym77** [comment]
Call for review ended. Merging.

**Pull request merged**